tests/unit/dataactvalidator/validation_handlers/test_file_generation_manager.py

```diff
@@ -23,6 +23,113 @@
     PublishedAwardFinancialAssistanceFactory)
 
 
+def test_tas_concats():
+    # everything
+    tas_dict = {
+        'allocation_transfer_agency': '097',
+        'agency_identifier': '017',
+        'beginning_period_of_availa': '2017',
+        'ending_period_of_availabil': '2017',
+        'availability_type_code': 'D',
+        'main_account_code': '0001',
+        'sub_account_code': '001'
+    }
+    tas2_str = concat_tas_dict(tas_dict)
+    tas2_dstr = concat_display_tas_dict(tas_dict)
+    assert tas2_str == '09701720172017D0001001'
+    assert tas2_dstr == '097-017-D-0001-001'
+
+    # everything sans type code
+    tas_dict = {
+        'allocation_transfer_agency': '097',
+        'agency_identifier': '017',
+        'beginning_period_of_availa': '2017',
+        'ending_period_of_availabil': '2017',
+        'availability_type_code': None,
+        'main_account_code': '0001',
+        'sub_account_code': '001'
+    }
+    tas2_str = concat_tas_dict(tas_dict)
+    tas2_dstr = concat_display_tas_dict(tas_dict)
+    assert tas2_str == '09701720172017 0001001'
+    assert tas2_dstr == '097-017-2017/2017-0001-001'
+
+    # everything sans ata
+    tas_dict = {
+        'allocation_transfer_agency': None,
+        'agency_identifier': '017',
+        'beginning_period_of_availa': '2017',
+        'ending_period_of_availabil': '2017',
+        'availability_type_code': 'D',
+        'main_account_code': '0001',
+        'sub_account_code': '001'
+    }
+    tas2_str = concat_tas_dict(tas_dict)
+    tas2_dstr = concat_display_tas_dict(tas_dict)
+    assert tas2_str == '00001720172017D0001001'
+    assert tas2_dstr == '017-D-0001-001'
+
+    # everything sans aid
+    tas_dict = {
+        'allocation_transfer_agency': '097',
+        'agency_identifier': None,
+        'beginning_period_of_availa': '2017',
+        'ending_period_of_availabil': '2017',
+        'availability_type_code': 'D',
+        'main_account_code': '0001',
+        'sub_account_code': '001'
+    }
+    tas2_str = concat_tas_dict(tas_dict)
+    tas2_dstr = concat_display_tas_dict(tas_dict)
+    assert tas2_str == '09700020172017D0001001'
+    assert tas2_dstr == '097-D-0001-001'
+
+    # everything sans periods
+    tas_dict = {
+        'allocation_transfer_agency': '097',
+        'agency_identifier': '017',
+        'beginning_period_of_availa': None,
+        'ending_period_of_availabil': None,
+        'availability_type_code': None,
+        'main_account_code': '0001',
+        'sub_account_code': '001'
+    }
+    tas2_str = concat_tas_dict(tas_dict)
+    tas2_dstr = concat_display_tas_dict(tas_dict)
+    assert tas2_str == '09701700000000 0001001'
+    assert tas2_dstr == '097-017-0001-001'
```

Comment:
Can we get a test showing what happens if we're missing just one of the dates? I know it isn't valid but during validation it's possible we'll have one like that, which is something the validations would catch but we'll still have to display it in the error/warning reports